hs_core/views/__init__.py

```diff
@@ -917,38 +917,22 @@ def _share_resource(request, shortkey, privilege, user_or_group_id, user_or_grou
     else:
         status = 'error'
 
-    current_user_privilege = res.raccess.get_effective_privilege(user)
-    if current_user_privilege == PrivilegeCodes.VIEW:
-        current_user_privilege = "view"
-    elif current_user_privilege == PrivilegeCodes.CHANGE:
-        current_user_privilege = "change"
-    elif current_user_privilege == PrivilegeCodes.OWNER:
-        current_user_privilege = "owner"
+    from hs_core.models import get_user_object
 
     if user_or_group == 'user':
-        is_current_user = False
-        if user == user_to_share_with:
-            is_current_user = True
-
-        picture_url = None
-        if user_to_share_with.userprofile.picture:
-            picture_url = user_to_share_with.userprofile.picture.url
-
-        ajax_response_data = {'status': status, 'name': user_to_share_with.get_full_name(),
-                              'username': user_to_share_with.username, 'privilege_granted': privilege,
-                              'current_user_privilege': current_user_privilege,
-                              'profile_pic': picture_url, 'is_current_user': is_current_user,
-                              'error_msg': err_message}
+        user_can_undo = request.user.uaccess.can_undo_share_resource_with_user(res, user_to_share_with)
+        user_to_share_with.can_undo = user_can_undo
 
+        ajax_response_data = {'status': status,
+                              'error_msg': err_message,
+                              'user': get_user_object(user_to_share_with, "user", privilege)}
     else:
-        group_pic_url = None
-        if group_to_share_with.gaccess.picture:
-            group_pic_url = group_to_share_with.gaccess.picture.url
-
-        ajax_response_data = {'status': status, 'name': group_to_share_with.name,
-                              'privilege_granted': privilege, 'group_pic': group_pic_url,
-                              'current_user_privilege': current_user_privilege,
-                              'error_msg': err_message}
+        group_can_undo = request.user.uaccess.can_undo_share_resource_with_group(res, group_to_share_with)
+        group_to_share_with.can_undo = group_can_undo
+
+        ajax_response_data = {'status': status,
+                              'error_msg': err_message,
+                              'user': get_user_object(group_to_share_with, "group", privilege)}
```

Comment:
Could you separate group and user?  It's confusing to smash group into the user model with a different type, it wasn't like this originally.

Reply:
Could you elaborate?

Reply:
you have a user key and a method that named get_user_object... the method will return either a user or a group.  And then it will be placed in a key with user.

Reply:
Understood. I'll use another key in case of Groups so it's clear what is being returned.

Reply:
I chose to use a more appropriate method name instead. The object returned represent an access entry in the resource.